python/python/lance/dependencies.py

```diff
@@ -0,0 +1,211 @@
+# Originally from https://github.com/pola-rs/polars/blob/ea4389c31b0e87ddf20a85e4c3797b285966edb6/py-polars/polars/dependencies.py
+# License: MIT, Copyright (c) 2020 Ritchie Vink
+# https://github.com/pola-rs/polars/blob/main/LICENSE
+
+from __future__ import annotations
+
+import re
+import sys
+from functools import lru_cache
+from importlib import import_module
+from importlib.util import find_spec
+from types import ModuleType
+from typing import TYPE_CHECKING, Any, ClassVar, Hashable, cast
+
+_NUMPY_AVAILABLE = True
+_PANDAS_AVAILABLE = True
+_POLARS_AVAILABLE = True
+_TORCH_AVAILABLE = True
+
+
+class _LazyModule(ModuleType):
+    """
+    Module that can act both as a lazy-loader and as a proxy.
+
+    Notes
+    -----
+    We do NOT register this module with `sys.modules` so as not to cause
+    confusion in the global environment. This way we have a valid proxy
+    module for our own use, but it lives _exclusively_ within lance.
+
+    """
+
+    __lazy__ = True
+
+    _mod_pfx: ClassVar[dict[str, str]] = {
+        "numpy": "np.",
+        "pandas": "pd.",
+        "polars": "pl.",
+    }
```

Comment:
Does torch need to be in this list as `"torch": "torch."` or can it be excluded since it isn't really a prefix?

Reply:
Good catch. I will add the one for `torch`. The prefixes are basically for displaying missing imports. For instance, when I test locally, it would say the following if the `pd` prefix is registered:

<pre>ModuleNotFoundError: <b>pd</b>.DataFrame requires 'pandas' module to be installed
</pre>

versus the following if the `pd` prefix is registered:

<pre>ModuleNotFoundError: DataFrame requires 'pandas' module to be installed`
</pre>